pkl-cli/src/main/kotlin/org/pkl/cli/CliFormatterCommand.kt

```diff
@@ -15,40 +15,46 @@
  */
 package org.pkl.cli
 
+import com.github.ajalt.clikt.core.ProgramResult
+import java.io.IOException
 import java.io.Writer
+import java.nio.file.Files
 import java.nio.file.Path
 import kotlin.io.path.ExperimentalPathApi
 import kotlin.io.path.extension
 import kotlin.io.path.isDirectory
 import kotlin.io.path.name
 import kotlin.io.path.walk
+import kotlin.io.path.writeText
 import org.pkl.commons.cli.CliBaseOptions
 import org.pkl.commons.cli.CliCommand
 import org.pkl.formatter.Formatter
 import org.pkl.formatter.GrammarVersion
 import org.pkl.parser.GenericParserError
 
-abstract class CliFormatterCommand
+class CliFormatterCommand
 @JvmOverloads
 constructor(
-  options: CliBaseOptions,
-  protected val paths: List<Path>,
-  protected val grammarVersion: GrammarVersion,
-  protected val consoleWriter: Writer = System.out.writer(),
-) : CliCommand(options) {
-  protected fun format(file: Path, contents: String): Pair<String, Int> {
-    try {
-      return Formatter().format(contents, grammarVersion) to 0
-    } catch (pe: GenericParserError) {
-      consoleWriter.write("Could not format `$file`: $pe")
-      consoleWriter.appendLine()
-      consoleWriter.flush()
-      return "" to 1
-    }
+  private val paths: List<Path>,
+  private val grammarVersion: GrammarVersion,
+  private val overwrite: Boolean,
+  private val names: Boolean,
+  private val consoleWriter: Writer = System.out.writer(),
+  private val errWriter: Writer = System.err.writer(),
+) : CliCommand(CliBaseOptions()) {
+
+  private fun format(contents: String): String {
+    return Formatter().format(contents, grammarVersion)
+  }
+
+  private fun writeErr(error: String) {
+    errWriter.write(error)
+    errWriter.appendLine()
+    errWriter.flush()
   }
 
   @OptIn(ExperimentalPathApi::class)
-  protected fun paths(): Set<Path> {
+  private fun paths(): Set<Path> {
```

Comment:
We can improve this a bit by producing a `Stream` here (don't wait until all trees are walked before formatting starts).

Also, produce normalized and absolute paths so that our errors don't look something like: `/Users/foo/./bar/baz.pkl`.

Also, nit: don't collide method name with field name `paths`

```kt
  private fun allPaths(): Stream<Path> {
    return paths
      .distinct()
      .stream()
      .flatMap { path ->
        if (path.isDirectory()) {
          Files.walk(path).filter { it.extension == "pkl" || it.name == "PklProject" }
        } else {
          Stream.of(path)
        }
      }
      .map { it.normalize().toAbsolutePath() }
  }
```